homeassistant/components/emulated_roku/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow to configure emulated_roku component."""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.util import slugify
+from .const import (DOMAIN, CONF_LISTEN_PORT, CONF_HOST_IP,
+                    CONF_ADVERTISE_IP, CONF_ADVERTISE_PORT,
+                    CONF_UPNP_BIND_MULTICAST,
+                    DEFAULT_UPNP_BIND_MULTICAST)
+
+
+@callback
+def configured_servers(hass):
+    """Return a set of the configured servers."""
+    return set(slugify(entry.data[CONF_NAME]) for entry
+               in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class EmulatedRokuFlowHandler(config_entries.ConfigFlow):
+    """Handle an emulated_roku config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    _hassio_discovery = None
+
+    def __init__(self):
+        """Initialize emulated_roku configuration flow."""
+        pass
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            name = slugify(user_input[CONF_NAME])
+
+            if name in configured_servers(self.hass):
+                return self.async_abort(reason='name_exists')
+
+            return self.async_create_entry(
+                title=user_input[CONF_NAME],
+                data=user_input
+            )
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema({
+                vol.Required(CONF_NAME): str,
+                vol.Required(CONF_LISTEN_PORT): vol.Coerce(int),
+                vol.Optional(CONF_HOST_IP): str,
```

Comment:
what are all these options and why are they needed, they seem to be like advanced options.

Config flows should only gather data that a normal user would need. All others need defaults and only be configured via configuration.yaml

Reply:
As with emulated_hue, bound ports/addresses might be different because of docker.
BTW, there is no configuration.yaml, so if I remove it from the UI, it's only editable by editing the JSON I guess?

Only name / port is required, others are optional and the third party lib handles if they are missing. (host is handled by the component like emulated_hue)

Reply:
Config flows need to be as simple as possible. It's meant for normal users. All these advanced options should be moved to configuration.yaml. See for example MQTT on how to combine config entries with configuration.yaml config.

I think that this config entry should just be a confirm and then just start. All other options via configuration.yaml.

Reply:
I see.
The earlier versions had both config entries and yaml, but I intentionally removed the .yaml path (see the discussion above, I think it's confusing to import the config from the yaml to the JSON for a brand new component)

Reply:
Added yaml support back without any defaults; only name and listen_port is required, missing host_ip is handled by the component, rest of the properties are completely optional.
The yaml entries are imported as config entries.

Removed all but required options (which now have defaults) from the config flow.